### PR TITLE
feat: ingest pyarrow span evals into sqlite

### LINKED ISSUE #2778: [persistence] pyarrow to sqlite ingestion of evals
Take pyarrow file ingests and perform bulk inserts on annotation tables. Note that this can cause permance issues so needs to be stress tested.

### PR DESCRIPTION
**Pull request by axiomofjoy**
adds sqlite support for span evals ingested via pyarrow

resolves #2778

### CODE REVIEW

File: src/phoenix/server/app.py [hunk 1]
```diff
@@ -182,8 +182,11 @@ def _lifespan(
 ) -> StatefulLifespan[Starlette]:
     @contextlib.asynccontextmanager
     async def lifespan(_: Starlette) -> AsyncIterator[Dict[str, Any]]:
-        async with BulkInserter(db, initial_batch_of_spans) as queue_span:
-            yield {"queue_span_for_bulk_insert": queue_span}
+        async with BulkInserter(db, initial_batch_of_spans) as (queue_span, queue_evaluation):
```
Comment (RogerHYang) [inline src/phoenix/server/app.py:185]: can we also pipe in the initial evals from the fixture?
```suggestion
        async with BulkInserter(db, initial_batch_of_spans, inital_batch_of_evaluations) as (queue_span, queue_evaluation):
```

File: src/phoenix/server/api/routers/evaluation_handler.py [hunk 2]
```diff
@@ -100,11 +100,15 @@ async def _process_pyarrow(self, request: Request, project_name: str) -> Respons
...
                 self._add_evaluations,
+                request,
                 evaluations,
                 project_name,
             )
         )
 
-    async def _add_evaluations(self, evaluations: Evaluations, project_name: str) -> None:
+    async def _add_evaluations(
+        self, request: Request, evaluations: Evaluations, project_name: str
```
Comment (RogerHYang) [inline src/phoenix/server/api/routers/evaluation_handler.py:110]: nit. maybe use the State instead, so it's a bit more explicit
```suggestion
        self, state: State, evaluations: Evaluations, project_name: str
```

**Review by RogerHYang — APPROVED** [review]

File: src/phoenix/server/api/routers/evaluation_handler.py (hunk 2 above)
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/evaluation_handler.py:110]: Good idea

File: src/phoenix/db/bulk_inserter.py [hunk 3]
```diff
@@ -75,6 +84,46 @@ async def _insert_spans(self) -> None:
...
+                logger.exception("Failed to insert evaluations")
+
+
+async def _insert_evaluation(session: AsyncSession, evaluation: pb.Evaluation) -> None:
+    if not (
+        span_rowid := await session.scalar(
+            select(models.Span.id).where(models.Span.span_id == evaluation.subject_id.span_id)
+        )
+    ):
+        return
```
Comment (RogerHYang) [inline src/phoenix/db/bulk_inserter.py:112]: maybe this should be an exception
```suggestion
        raise ValueError(f"Span not found: {span_id=}")
```

Comment (axiomofjoy) [reply src/phoenix/db/bulk_inserter.py:112]: I don't think `session.begin_nested` is working as expected. When I raise an exception here, it kills the entire transaction and no subsequent evals are inserted. Going to leave as a `return` for now and file and issue to fix.

File: src/phoenix/server/api/routers/evaluation_handler.py (hunk 2 above)
Comment (axiomofjoy) [reply src/phoenix/server/api/routers/evaluation_handler.py:110]: addressed in ae16f08aa3e1222fef7d7bd379e5a7bde41b88ea

File: src/phoenix/server/app.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/server/app.py:185]: I need to first add support for other kinds of evaluations (e.g., document and trace evaluations). Filing this as a ticket.

**Pull request merged**